### PR TITLE
AppBar: fix bugs, add docs, add samples

### PR DESCRIPTION
**Pull request by Hixie**
I added some tests for the bug that I fixed.
I added docs for IconButton and AppBar.
I added some new constructors for FractionalOffset.

### CODE REVIEW

**Comment by Hixie** [comment]
cc @abarth

**Comment by Hixie** [comment]
Depends on https://github.com/flutter/assets-for-api-docs/pull/4

File: packages/flutter/lib/src/painting/fractional_offset.dart [hunk 1]
```diff
@@ -8,10 +8,11 @@ import 'package:flutter/foundation.dart';
...
 import 'basic_types.dart';
 
-/// An offset that's expressed as a fraction of a Size.
+/// An offset that's expressed as a fraction of a [Size].
 ///
-/// FractionalOffset(1.0, 0.0) represents the top right of the Size,
-/// FractionalOffset(0.0, 1.0) represents the bottom left of the Size,
+/// `FractionalOffset(1.0, 0.0)` represents the top right of the [Size].
+///
+/// `FractionalOffset(0.0, 1.0)` represents the bottom left of the [Size].
```
Comment (xster) [inline packages/flutter/lib/src/painting/fractional_offset.dart]: maybe also mention that the values can be <0, >1 to represent offscreen/outside [size] locations

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/painting/fractional_offset.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/painting/fractional_offset.dart]: Done.

File: packages/flutter/lib/src/material/app_bar.dart [hunk 2]
```diff
@@ -119,17 +119,53 @@ class _ToolbarContainerLayout extends SingleChildLayoutDelegate {
 ///
 /// An app bar consists of a toolbar and potentially other widgets, such as a
 /// [TabBar] and a [FlexibleSpaceBar]. App bars typically expose one or more
-/// common actions with [IconButton]s which are optionally followed by a
+/// common [actions] with [IconButton]s which are optionally followed by a
```
Comment (xster) [inline packages/flutter/lib/src/material/app_bar.dart]: followed by more actions folded in a [PopupMenuButton] overflow for less common options?

Think it's good to draw reference to 'overflow' so people click on common terminology right away

File: packages/flutter/lib/src/material/app_bar.dart [hunk 3]
```diff
@@ -119,17 +119,53 @@ class _ToolbarContainerLayout extends SingleChildLayoutDelegate {
...
 /// App bars are typically used in the [Scaffold.appBar] property, which places
 /// the app bar as a fixed-height widget at the top of the screen. For a
 /// scrollable app bar, see [SliverAppBar], which embeds an [AppBar] in a sliver
 /// for use in a [CustomScrollView].
 ///
-/// The AppBar displays the toolbar widgets, [leading], [title], and
-/// [actions], above the [bottom] (if any). If a [flexibleSpace] widget is
-/// specified then it is stacked behind the toolbar and the bottom widget.
+/// The AppBar displays the toolbar widgets, [leading], [title], and [actions],
+/// above the [bottom] (if any). If a [flexibleSpace] widget is specified then
```
Comment (xster) [inline packages/flutter/lib/src/material/app_bar.dart]: I have trouble visualizing what purpose/common use pattern of the bottom is for just reading this and the next paragraphs. Maybe add a half sentence elaboration.

File: packages/flutter/lib/src/material/icon_button.dart [hunk 4]
```diff
@@ -37,6 +37,16 @@ const double _kMinButtonSize = 48.0;
 /// requirements in the Material Design specification. The [alignment] controls
 /// how the icon itself is positioned within the hit region.
 ///
+/// ## Sample code
+///
+/// ```dart
+/// new IconButton(
```
Comment (xster) [inline packages/flutter/lib/src/material/icon_button.dart]: +100000

**Review by xster — APPROVED** [review]
LG!

File: packages/flutter/lib/src/material/app_bar.dart (hunk 2 above)
Comment (Hixie) [inline packages/flutter/lib/src/material/app_bar.dart]: Added `(sometimes called the "overflow menu")`.

At some point we should probably do the folding into the menu thing automatically.

File: packages/flutter/lib/src/material/app_bar.dart (hunk 3 above)
Comment (Hixie) [inline packages/flutter/lib/src/material/app_bar.dart]: Added `The [bottom] is usually used for a [TabBar].` after `(if any).`.

**Comment by Hixie** [comment]
will land on green

**Pull request merged**